### ISSUE TITLE
Loss of precision when formatting number

### ISSUE THREAD

**Issue by markddrake** [molsonkiko/JsonToolsNppPlugin#78]
```
{"systemInformation":{"yadamuVersion":"3.0","date":"2024-08-07T01:49:47.278Z","timeZoneOffset":420,"schema":"t_postgres","dbiKey":"postgres","vendor":"Postgres","dbVersion":"160001","softwareVendor":"The PostgreSQL Global Development Group","driverSettings":{"spatialFormat":"EWKB","timestampPrecision":6,"circleFormat":"CIRCLE"},"nodeClient":{"version":"v20.7.0","architecture":"x64","platform":"win32"},"currentUser":"postgres","sessionUser":"postgres","dbName":"yadamu","databaseVersion":"160001","timezone":"+00:00","postgisInfo":"3.4 USE_GEOS=1 USE_PROJ=1 USE_STATS=1","yadamuInstanceID":"5F0997C3-8D35-484C-9330-7D40370A6D39","yadamuInstallationTimestamp":"2024-08-02T21:06:18+00:00"},"ddl":[],"metadata":{"t1":{"tableSchema":"t_postgres","tableName":"t1","columnNames":["key","val"],"dataTypes":["smallint","double precision"],"sizeConstraints":[[16,0],[53]],"vendor":"Postgres"}},"data":{"t1":[[1,-1.7976931348623157e+308],[2,1.7976931348623157e+308],[26,-1.7976931348623157e+308],[27,1.7976931348623157e+308]]}}
```
Becomes
```
{
    "systemInformation": {
        "yadamuVersion": "3.0",
        "date": "2024-08-07T01:49:47.278Z",
        "timeZoneOffset": 420,
        "schema": "t_postgres",
        "dbiKey": "postgres",
        "vendor": "Postgres",
        "dbVersion": "160001",
        "softwareVendor": "The PostgreSQL Global Development Group",
        "driverSettings": {
            "spatialFormat": "EWKB",
            "timestampPrecision": 6,
            "circleFormat": "CIRCLE"
        },
        "nodeClient": {
            "version": "v20.7.0",
            "architecture": "x64",
            "platform": "win32"
        },
        "currentUser": "postgres",
        "sessionUser": "postgres",
        "dbName": "yadamu",
        "databaseVersion": "160001",
        "timezone": "+00:00",
        "postgisInfo": "3.4 USE_GEOS=1 USE_PROJ=1 USE_STATS=1",
        "yadamuInstanceID": "5F0997C3-8D35-484C-9330-7D40370A6D39",
        "yadamuInstallationTimestamp": "2024-08-02T21:06:18+00:00"
    },
    "ddl": [
    ],
    "metadata": {
        "t1": {
            "tableSchema": "t_postgres",
            "tableName": "t1",
            "columnNames": [
                "key",
                "val"
            ],
            "dataTypes": [
                "smallint",
                "double precision"
            ],
            "sizeConstraints": [
                [
                    16,
                    0
                ],
                [
                    53
                ]
            ],
            "vendor": "Postgres"
        }
    },
    "data": {
        "t1": [
            [
                1,
                -1.79769313486232E+308
            ],
            [
                2,
                1.79769313486232E+308
            ],
            [
                26,
                -1.79769313486232E+308
            ],
            [
                27,
                1.79769313486232E+308
            ]
        ]
    }
}
```

Does not happen with JSON_VIEWER plugin

```
{
	"systemInformation": {
		"yadamuVersion": "3.0",
		"date": "2024-08-07T01:49:47.278Z",
		"timeZoneOffset": 420,
		"schema": "t_postgres",
		"dbiKey": "postgres",
		"vendor": "Postgres",
		"dbVersion": "160001",
		"softwareVendor": "The PostgreSQL Global Development Group",
		"driverSettings": {
			"spatialFormat": "EWKB",
			"timestampPrecision": 6,
			"circleFormat": "CIRCLE"
		},
		"nodeClient": {
			"version": "v20.7.0",
			"architecture": "x64",
			"platform": "win32"
		},
		"currentUser": "postgres",
		"sessionUser": "postgres",
		"dbName": "yadamu",
		"databaseVersion": "160001",
		"timezone": "+00:00",
		"postgisInfo": "3.4 USE_GEOS=1 USE_PROJ=1 USE_STATS=1",
		"yadamuInstanceID": "5F0997C3-8D35-484C-9330-7D40370A6D39",
		"yadamuInstallationTimestamp": "2024-08-02T21:06:18+00:00"
	},
	"ddl": [],
	"metadata": {
		"t1": {
			"tableSchema": "t_postgres",
			"tableName": "t1",
			"columnNames": [
				"key",
				"val"
			],
			"dataTypes": [
				"smallint",
				"double precision"
			],
			"sizeConstraints": [
				[
					16,
					0
				],
				[
					53
				]
			],
			"vendor": "Postgres"
		}
	},
	"data": {
		"t1": [
			[
				1,
				-1.7976931348623157e308
			],
			[
				2,
				1.7976931348623157e308
			],
			[
				26,
				-1.7976931348623157e308
			],
			[
				27,
				1.7976931348623157e308
			]
		]
	}
}
```

**Comment by molsonkiko** [comment]
@markddrake 

Hmmm, unfortunately I can replicate this bug with the latest version of JsonTools. Thanks for bringing it to my attention.

This appears to be a known issue with the way C# formats double-precision floating point numbers (which I use to represent the `number` type in JSON). I'm not yet sure how I'm going to fix this bug, but I will make sure to include a fix in version 8.1 of JsonTools.

**Comment by molsonkiko** [comment]
This issue should be resolved by [this commit](https://github.com/molsonkiko/JsonToolsNppPlugin/commit/7522a208e7af0804ff8c928050fbd65ba6cf26ec).

Again, sorry for not recognizing this issue sooner. I plan to release version 8.1 on the Plugins Manager fairly soon, and that version will include this fix.

**Comment by molsonkiko** [comment]
The fix to this issue is now in [v8.1](https://github.com/molsonkiko/JsonToolsNppPlugin/releases/tag/v8.1), which I will try to get listed in the Notepad++ Plugin List ASAP.

Once the next version of Notepad++ is released that lists that version of JsonTools, I will wait a little while and then close this issue.
